core/models/asset.py

```diff
@@ -229,6 +229,8 @@ def save(self, *args, **kwargs):
 def set_current_asset_status(sender, **kwargs):
     asset_status = kwargs.get('instance')
     asset_status.asset.current_status = asset_status.current_status
+    if asset_status.current_status == "Available":
+        asset_status.asset.assigned_to = None
     asset_status.asset.save()
```

Comment:
Shouldn't this also set the former `current_owner` to be the new `previous_owner`?

Reply:
I think @wanjiku-carol's comment makes sense